src/catalog/catalog.cpp

```diff
@@ -189,507 +184,406 @@ ResultType Catalog::CreateDatabase(const std::string &database_name,
-  if (!found_database) {
-    LOG_TRACE("Database %d is not found!", database_oid);
-    return ResultType::FAILURE;
-  }
   return ResultType::SUCCESS;
+  // Instead of dropping actual database object
+  // LOG_TRACE("Dropping database with oid: %d", database_oid);
+  // std::lock_guard<std::mutex> lock(catalog_mutex);
+  // bool found_database =
+  //     storage_manager->RemoveDatabaseFromStorageManager(database_oid);
+  // if (!found_database) {
+  //   LOG_TRACE("Database %d is not found!", database_oid);
+  //   return ResultType::FAILURE;
+  // }
+  // return ResultType::SUCCESS;
 }
 
 /*@brief   Drop table
-* 1. drop all the indexes on actual table, and drop index records in pg_index
-* 2. drop all the columns records in pg_attribute
-* 3. drop table record in pg_table
-* 4. delete actual table(storage level), cleanup schema, foreign keys,
-* tile_groups
-* @param   database_name    the database which the dropped table belongs to
-* @param   table_name       the dropped table name
-* @param   txn              Transaction
-* @return  Transaction ResultType(SUCCESS or FAILURE)
-*/
+ * 1. drop all the indexes on actual table, and drop index records in pg_index
+ * 2. drop all the columns records in pg_attribute
+ * 3. drop table record in pg_table
+ * 4. delete actual table(storage level), cleanup schema, foreign keys,
+ * tile_groups
+ * @param   database_name    the database which the dropped table belongs to
+ * @param   table_name       the dropped table name
+ * @param   txn              Transaction
+ * @return  Transaction ResultType(SUCCESS or FAILURE)
+ */
 ResultType Catalog::DropTable(const std::string &database_name,
                               const std::string &table_name,
                               concurrency::Transaction *txn) {
-  if (txn == nullptr) {
-    LOG_TRACE("Do not have transaction to drop table: %s", table_name.c_str());
-    return ResultType::FAILURE;
-  }
+  if (txn == nullptr)
+    throw CatalogException("Do not have transaction to drop table " +
+                           table_name);
 
   // Checking if statement is valid
-  oid_t database_oid =
-      DatabaseCatalog::GetInstance()->GetDatabaseOid(database_name, txn);
-  if (database_oid == INVALID_OID) {
-    LOG_TRACE("Cannot find database  %s!", database_name.c_str());
-    return ResultType::FAILURE;
-  }
-
-  oid_t table_oid =
-      TableCatalog::GetInstance()->GetTableOid(table_name, database_oid, txn);
-  if (table_oid == INVALID_OID) {
-    LOG_TRACE("Cannot find Table %s to drop!", table_name.c_str());
+  auto database_object =
+      DatabaseCatalog::GetInstance()->GetDatabaseObject(database_name, txn);
+  if (database_object == nullptr)
+    throw CatalogException("Drop Table: " + database_name + " does not exist");
+
+  // check if table exists
+  auto table_object = database_object->GetTableObject(table_name);
+  if (table_object == nullptr) {
+    // drop table if exists foo;
     return ResultType::FAILURE;
```

Comment:
Shouldn't you also throw an exception here? (to make it consistent with the rest of the function)